### PR TITLE
Feature/validate token with signincanada

### LINKED ISSUE #1146: API - Validate token with public key retrieved from jwk endpoint
# Description
When using our own auth service, we can save the public key in an env variable, and simply use that to validate tokens on the API service (in AuthServiceProvider).

When using an external auth service, we want to retrieve the public key from the openid jwk endpoint. For reference, Sign in Canada's keys are at https://te-auth.id.tbs-sct.gc.ca/oxauth/restv1/jwks.

I think the best way to do this without breaking compatibility with our own auth service is to use the env variable public key if it is present and non-null, otherwise we get it from a public key url env variable, otherwise we try to find the url using the `{AUTH_ISS}/.well_known` endpoint. 

QUESTION: the public key must also be parsed from the url, based on the algorithm used, and the format might need some transformation?

How do test this locally?

### PR DESCRIPTION
**Pull request by petertgiles**
This PR allows the admin site to log users in using the User Acceptance auth server for Sign In Canada: https://te-auth.id.tbs-sct.gc.ca/oxauth.  The site will also validate the issued tokens using the data from the jwks endpoint.

It code abstracts the auth configuration and validation behind an interface to allow easy configuration switching between local auth and SiC depending on the .env settings.

The tokens that are issued by SiC only last five minutes so we'll need to work on automatic requests of refresh tokens soon.  I'll submit an issue for that.

Also, the tokens issues by SiC require a uuid sub field unlike the email sub fields we are currently using.  For testing I suggest decoding the returned token with https://jwt.io/ to extract the sub field and using adminer to apply it manually to a user.

Closes #1146

### CODE REVIEW

**Comment by petertgiles** [comment]
Issue #1192 opened to keep session alive.
Issue #1193 opened to register SiC user.

File: admin/resources/js/components/AuthContainer.tsx [hunk 1]
```diff
@@ -44,7 +44,12 @@ export const AuthContainer: React.FC = ({ children }) => {
     const queryParams = parseUrlQueryParameters(location);
     const accessToken = queryParams.access_token;
     const refreshToken = queryParams.refresh_token;
-    if (accessToken && queryParams.token_type === "Bearer") {
+    if (
+      accessToken &&
+      queryParams.token_type.localeCompare("Bearer", "en", {
+        sensitivity: "base",
+      }) === 0
+    ) {
```
Comment (petertgiles) [inline admin/resources/js/components/AuthContainer.tsx:52]: This change will conflict with #1196.  This change is simpler than Tristan's and his handles the case insensitive to this one should probably be removed.

Comment (petertgiles) [reply admin/resources/js/components/AuthContainer.tsx]: OK, I merged in the newer AuthContainer.tsx file to resolve the conflict.

**Review by gggrant — APPROVED** [review]
Authentication through SiC working as described.

**Pull request merged**